src/common/security/local/context.go

```diff
@@ -89,109 +90,34 @@ func (s *SecurityContext) Can(action types.Action, resource types.Resource) bool
 		if s.IsSysAdmin() {
 			evaluators = evaluators.Add(admin.New(s.GetUsername()))
 		}
-		evaluators = evaluators.Add(rbac.NewProjectRBACEvaluator(s, s.pm))
+		evaluators = evaluators.Add(newEvaluator(s, s.pm))
 
 		s.evaluator = evaluators
 	})
 
 	return s.evaluator != nil && s.evaluator.HasPermission(resource, action)
 }
 
-// GetProjectRoles ...
-func (s *SecurityContext) GetProjectRoles(projectIDOrName interface{}) []int {
-	if !s.IsAuthenticated() || projectIDOrName == nil {
-		return []int{}
-	}
-
-	roles := []int{}
-	user, err := dao.GetUser(models.User{
-		Username: s.GetUsername(),
-	})
-	if err != nil {
-		log.Errorf("failed to get user %s: %v", s.GetUsername(), err)
-		return roles
-	}
-	if user == nil {
-		log.Debugf("user %s not found", s.GetUsername())
-		return roles
-	}
-	project, err := s.pm.Get(projectIDOrName)
-	if err != nil {
-		log.Errorf("failed to get project %v: %v", projectIDOrName, err)
-		return roles
-	}
-	if project == nil {
-		log.Errorf("project %v not found", projectIDOrName)
-		return roles
-	}
-	roleList, err := dao.GetUserProjectRoles(user.UserID, project.ProjectID, common.UserMember)
-	if err != nil {
-		log.Errorf("failed to get roles of user %d to project %d: %v", user.UserID, project.ProjectID, err)
-		return roles
-	}
-	for _, role := range roleList {
-		switch role.RoleCode {
-		case "MDRWS":
-			roles = append(roles, common.RoleProjectAdmin)
-		case "DRWS":
-			roles = append(roles, common.RoleMaster)
-		case "RWS":
-			roles = append(roles, common.RoleDeveloper)
-		case "RS":
-			roles = append(roles, common.RoleGuest)
-		case "LRS":
-			roles = append(roles, common.RoleLimitedGuest)
+func newEvaluator(ctx *SecurityContext, pm promgr.ProjectManager) evaluator.Evaluator {
```

Comment:
Suggest to use this function to replace `rbac.NewProjectRBACEvaluator`

Reply:
+1